cranelift/wasm/src/environ/spec.rs

```diff
@@ -112,11 +117,14 @@ pub trait FuncEnvironment: TargetEnvironment {
         index: GlobalIndex,
     ) -> WasmResult<GlobalVariable>;
 
+    /// Get the heaps for this function environment.
+    fn heaps(&self) -> &PrimaryMap<Heap, HeapData>;
```

Comment:
A bit more documentation might be good here; otherwise this API is a bit perplexing at first glance (below there is `make_heap`, so the translator is telling the environment about heaps, but here we provide information about heaps back to the translator, which is kind of subtle).

Something like: "The returned map should provide heap format details (encoded in `HeapData`) for each `Heap` that was previously returned by `make_heap()`. The translator will first call `make_heap` for each Wasm memory, and then later when translating code, will invoke `heaps()` to learn how to access the environment's implementation of each memory."